node/directives/withCurrentProfile.ts

```diff
@@ -71,35 +59,53 @@ export class WithCurrentProfile extends SchemaDirectiveVisitor {
   }
 }
 
-function getCurrentProfileFromSession(
+async function getCurrentProfileFromSession(
   context: Context
-): Promise<CurrentProfile | null> {
+): Promise<ProfileInfos> {
   return getSession(context).then((currentSession) => {
     const session = currentSession as SessionFields
 
     if (!session || !session.id) {
       throw new ResolverError('Error fetching session data')
     }
 
-    const profile =
-      session.impersonate && session.impersonate.profile
-        ? session.impersonate.profile
-        : session.profile
+    const profile = session?.impersonate?.profile
+      ? session.impersonate.profile
+      : session.profile
 
-    return profile
-      ? ({
-          email: profile && profile.email,
-          userId: profile && profile.id,
-        } as CurrentProfile)
-      : null
+    return {
+      currentProfile: profile
+        ? ({ email: profile.email, userId: profile.id } as CurrentProfile)
```

Comment:
I think you need this to get the value from session:
```suggestion
        ? ({ email: profile.email.value, userId: profile.id.value } as CurrentProfile)
```

You can refer to this implementation for more info:

https://github.com/vtex/subscriptions-graphql/blob/705e3e1c00d6f4ce7ce5d3745eef451fe32c024e/node/services/vtexid.ts#L6

Reply:
Nope, don't need to access `value`